script.js

```diff
@@ -1,6 +1,290 @@
-function runProgram(){
+let allBrawlers = []
 
+function runProgram(){
+    fetchData();
     console.log('runProgram');
-    //your code goes here
+    const searchButton = document.querySelector('#search-button');
+    searchButton.addEventListener('click', showBrawler);
+    searchButton.addEventListener('click', showBrawlerRarity);
+    searchButton.addEventListener('click', showBrawlerImage);
+    searchButton.addEventListener('click', showBrawlerDescription);
+    searchButton.addEventListener('click', showBrawlerG1Name);
+    searchButton.addEventListener('click', showBrawlerG2Name);
+    searchButton.addEventListener('click', showBrawlerG1Description);
+    searchButton.addEventListener('click', showBrawlerG1Image);
+    searchButton.addEventListener('click', showBrawlerG2Description);
+    searchButton.addEventListener('click', showBrawlerG2Image);
+    searchButton.addEventListener('click', showBrawlerS1Name);
+    searchButton.addEventListener('click', showBrawlerS1Description);
+    searchButton.addEventListener('click', showBrawlerS1Image);
+    searchButton.addEventListener('click', showBrawlerS2Name);
+    searchButton.addEventListener('click', showBrawlerS2Description);
+    searchButton.addEventListener('click', showBrawlerS2Image);
```

Comment:
JP, you have a solid attempt at breaking up your problems into subproblems here, but perhaps you over-complicated things, or rather, you did not see that you could combine similar logic. 

If you look at all of your functions, you're repeating the same thing over and over. You could have done something like this: 

`searchButton.addEventListener('click', showBrawlerInfo)`

and that function would be something like:

``` javascript
 async function showBrawler(){
    const searchInput = document.querySelector('#search-input');
    const searchValue = searchInput.value;

for (const currBrawler of allBrawlers) {

    if(currBrawler.hash === searchValue){
         const BrawlerName = document.querySelector("#brawler-name");
         BrawlerName.textContent = currBrawler.hash;
         const BrawlerRarity = document.querySelector("#rarity");
         BrawlerRarity.textContent = currBrawler.rarity.name;
  }
}
}
```

This way, you would have avoided reading the search Input and looping through the currentBrawler every single time for every singly attribute. Hopefully, that makes sense.